numpy/core/arrayprint.py

```diff
@@ -650,18 +646,22 @@ def fillFormat(self, data):
                 format = '%'
             format = format + '%d.%de' % (self.max_str_len, self.precision)
         else:
-            format = '%%.%df' % (self.precision,)
             if len(non_zero):
-                precision = max([_digits(x, self.precision, format)
-                                 for x in non_zero])
+                precision = max([_trailing_nonzero(x, self.precision)
```

Comment:
No need to construct a list inside `max` -- btw, unrelated to this PR, but this seems a really slow way for large arrays...